lib/abilities.ex

```diff
@@ -0,0 +1,8 @@
+defimpl Canada.Can, for: Pairmotron.User do
+  alias Pairmotron.{PairRetro, User}
+
+  def can?(%User{id: user_id}, action, %PairRetro{user_id: user_id})
+    when action in [:edit, :update, :show, :delete], do: true
+
+  def can?(%User{}, _, _), do: false
+end
```

Comment:
The thought here is that canary will only be used on controller actions for now explicitly. If you use load_and_authorize_resource on something other than the above actions on the `PairRetroController`, you will have to whitelist it here.

Also of note, I am only using canary to authorize resources that can only be accessed by the associated user. At this time, this is only `PairRetro`s. This removed a ton of code from the pair retro controller actions, and made things much cleaner. It also removed a lot of duplicate code to popular and well maintained abstractions (`canada` and `canary`). The fact that they are well maintained and some of the go to solutions to authorizing specific resources mitigates the risk of adding these dependencies. They are also very lightweight compared to other solutions.